Tools/AP_Bootloader/bl_protocol.cpp

```diff
@@ -122,14 +122,28 @@
 #define PROTO_READ_MULTI_MAX    255	// size of the size field
 
 /* argument values for PROTO_GET_DEVICE */
-#define PROTO_DEVICE_BL_REV	1	// bootloader revision
-#define PROTO_DEVICE_BOARD_ID	2	// board ID
-#define PROTO_DEVICE_BOARD_REV	3	// board revision
-#define PROTO_DEVICE_FW_SIZE	4	// size of flashable area
-#define PROTO_DEVICE_VEC_AREA	5	// contents of reserved vectors 7-10
-#define PROTO_DEVICE_EXTF_SIZE  6   // size of available external flash
+#define PROTO_DEVICE_BL_REV       1  // bootloader revision
+#define PROTO_DEVICE_BOARD_ID	  2  // board ID
+#define PROTO_DEVICE_BOARD_REV	  3  // board revision
+#define PROTO_DEVICE_FW_SIZE	  4  // size of flashable area
+#define PROTO_DEVICE_VEC_AREA	  5  // contents of reserved vectors 7-10
+#define PROTO_DEVICE_EXTF_SIZE    6  // size of available external flash
+#define PROTO_DEVICE_BL_GIT_HASH  7  // git hash of bootloader build (requires > v5)
+
 // all except PROTO_DEVICE_VEC_AREA and PROTO_DEVICE_BOARD_REV should be done
-#define CHECK_GET_DEVICE_FINISHED(x)   ((x & (0xB)) == 0xB)
+
+// Convert the Device Info to a bitmask
+#define REQUIRE_DEVICE_INFO(x) (1<<(x-1))
```

Comment:
This should be used below (line 627) where `done_get_device_flags` is updated, i.e. change:
```
            done_get_device_flags |= (1<<(arg-1)); // set the flags for use when resetting timeout 
```
to:
```
            done_get_device_flags |= REQUIRE_DEVICE_INFO(arg); // set the flags for use when resetting timeout 
```